lib/tasks/publish_provider_check.rake

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+class PublishProviderChecker
+  include ServicePattern
+
+  PUBLISH_PROVIDER_ENDPOINT = "https://api.publish-teacher-training-courses.service.gov.uk/api/public/v1/recruitment_cycles/2024/providers"
```

Comment:
Perhaps have `2024` as parameter defaulting to `Setting.current_recruitment_cycle_year`